src/Table.ts

```diff
@@ -236,6 +236,7 @@ export default class Table {
             });
         }
 
+        const lastColumnIndex = (columns as ColumnDefinition[]).sort((a, b) => b.index - a.index)[0]?.index ?? 0;
```

Comment:
I'd adjust this line for two reasons:
* The `columns` parameter is a `ReadonlyArray` to avoid manipulating the Array and accidentally cause unwanted side effects for the caller of `getDataFromPage`. That's also why `.sort(...)` was not available. We need to make sure the array is copied and we do not work on the input parameter.
* The code suggestion I made expresses what we actually want to achieve with that line: finding the largest column index. Also, it is shorter and has less null handling.

```suggestion
        const lastColumnIndex = Math.max(...columns.map(c => c.index), 0)
```